asserts/database.go

```diff
@@ -89,6 +89,24 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// RevisionError indicates a revision improperly used for an operation.
+type RevisionError struct {
+	Used, Current int
+}
+
+func (e *RevisionError) Error() string {
+	if e.Used < 0 || e.Current < 0 {
+		return fmt.Sprintf("assertion revision is unknown")
```

Comment:
could add a TODO here that says something about the message may need tweaking once we hit a real use case for this one